File: test/python3/framing/test_newline.py
from unittest import TestCase

from .dummy_type_rxt import frame, unframe

class NewlineFramingTestCase(TestCase):

    def test_frame(self):
        packet = frame('abcdef')
        self.assertEqual('abcdef\n', packet)

    def test_frame_with_newline(self):
        with self.assertRaises(ValueError):
            frame('abc\ndef')

    def test_unframe_complete(self):
        context, packets = unframe('', 'abc\n')
        self.assertEqual(1, len(packets))
        self.assertEqual('abc', packets[0])

    def test_unframe_complete_with_context(self):
        context, packets = unframe('123', 'abc\n')
        self.assertEqual(1, len(packets))
        self.assertEqual('123abc', packets[0])

    def test_unframe_partial(self):
        context, packets = unframe('', 'abc\nefg')
        self.assertEqual(1, len(packets))
        self.assertEqual('abc', packets[0])
        self.assertEqual('efg', context)

    def test_unframe_partial_with_context(self):
        context, packets = unframe('iop', 'abc\nefg')
        self.assertEqual(1, len(packets))
        self.assertEqual('iopabc', packets[0])
        self.assertEqual('efg', context)

    def test_unframe_multiple(self):
        context, packets = unframe('iop', 'abc\nefg\nhij\nklm')
        self.assertEqual(3, len(packets))
        self.assertEqual('iopabc', packets[0])
        self.assertEqual('efg', packets[1])
        self.assertEqual('hij', packets[2])
        self.assertEqual('klm', context)

File: test/es2015/framing/test-newline.js
import {
  frame, unframe
} from './dummy_type_rxt.js';

var assert = require('assert');

describe('framing of', function() {
  describe('a string', function() {
    it('should append a newline', function() {
      let packet = frame('abcdef');
      assert.equal(packet, 'abcdef\n');
    });
  });
});

describe('framing of', function() {
  describe('a string containing a new line', function() {
    it('should raise an exception', function() {
      assert.throws(() => {
        frame('abc\ndef');
      });
    });
  });
});

describe('unframing of', function() {
  describe('a complete chunk', function() {
    it('should return a packet', function() {
      const result = unframe('', 'abc\n');
      assert.equal(1, result.packets.length);
      assert.equal('abc', result.packets[0]);
    });
  });
});

describe('unframing of', function() {
  describe('a complete chunk with a context', function() {
    it('should return a packet with context at the begining', function() {
      const result = unframe('123', 'abc\n');
      assert.equal(1, result.packets.length);
      assert.equal('123abc', result.packets[0]);
    });
  });
});

describe('unframing of', function() {
  describe('a partial chunk', function() {
    it('should return a packet and a context', function() {
      const result = unframe('', 'abc\nefg');
      assert.equal(1, result.packets.length);
      assert.equal('abc', result.packets[0]);
      assert.equal('efg', result.context);
    });
  });
});

describe('unframing of', function() {
  describe('a partial chunk with context', function() {
    it('should return a packet and a context', function() {
      const result = unframe('iop', 'abc\nefg');
      assert.equal(1, result.packets.length);
      assert.equal('iopabc', result.packets[0]);
      assert.equal('efg', result.context);
    });
  });
});

describe('unframing of', function() {
  describe('a chunk containing multiple packets', function() {
    it('should return the contained packets', function() {
      const result = unframe('iop', 'abc\nefg\nhij\nklm');
      assert.equal(3, result.packets.length);
      assert.equal('iopabc', result.packets[0]);
      assert.equal('efg', result.packets[1]);
      assert.equal('hij', result.packets[2]);
      assert.equal('klm', result.context);
    });
  });
});
